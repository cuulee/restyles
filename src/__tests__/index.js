/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import Style from '../index'

describe('cutest', () => {
  test('renders crazy stuff', () => {
    const tree = renderer
      .create(
        <Style
          css={{
            color: 'blue',
            '& .profile': {
              color: 'red',
              '& .username': {
                color: 'green',
                '& .three': {
                  color: 'gray',
                  '& .inner': {
                    color: 'rebeccapurple',
                    fontSize: 20
                  }
                }
              }
            },
            '&:hover': {
              '& .one': {
                color: 'blue',
                '& .two': {
                  color: 'gray',
                  '& .three': {
                    color: 'black',
                    '& .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            },
            '&:active': {
              '& .one': {
                color: 'blue',
                '& .two': {
                  color: 'gray',
                  '& .three': {
                    color: 'black',
                    '& .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            },
            '&:before': {
              '& .one': {
                color: 'blue',
                '& .two': {
                  color: 'gray',
                  '& .three': {
                    color: 'black',
                    '& .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            },
            '@media (min-width: 500px) and (orientation: landscape)': {
              '& .one': {
                color: 'blue',
                '& .two': {
                  color: 'gray',
                  '& .three': {
                    color: 'black',
                    '& .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            }
          }}
        >
          {rules => (
            <div {...rules}>
              blue
              <div className={'profile'}>
                red
                <div className={'username'}>
                  green
                  <div className={'three'}>
                    <span className={'inner'}>test</span>
                  </div>
                </div>
              </div>
            </div>
          )}
        </Style>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('use glamor helpers', () => {
    const tree = renderer
      .create(
        <Style
          css={[
            {color: 'lightgray'},
            {height: '100vh'},
            {
              color: 'blue',
              '&:hover': {
                background: 'blue'
              },
              '& .one': {
                color: 'red',
                '& .two': {
                  color: 'green',
                  '& .three': {
                    color: 'gray',
                    '& .inner': {
                      color: 'rebeccapurple',
                      fontSize: 20
                    }
                  }
                }
              }
            }
          ]}
        >
          {[
            rules => (
              <div {...rules}>
                <div className={'one'}>
                  <div className={'two'}>
                    <div className={'three'}>
                      <span className={'inner'}>test</span>
                    </div>
                  </div>
                </div>
              </div>
            ),
            rules => (
              <div {...rules}>
                <div className={'three'}>
                  <div className={'two'}>
                    <div className={'one'}>
                      <span className={'inner'}>test</span>
                    </div>
                  </div>
                </div>
              </div>
            )
          ]}
        </Style>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
