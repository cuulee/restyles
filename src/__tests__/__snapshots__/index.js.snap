// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cutest renders crazy stuff 1`] = `
<div
  data-css-oypqrq=""
>
  blue
  <div
    className="profile"
  >
    red
    <div
      className="username"
    >
      green
      <div
        className="three"
      >
        <span
          className="inner"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`cutest renders simple 1`] = `
<div
  data-css-epkc5v=""
>
  This will be blue until hovered.
  <div
    className="profile"
  >
    This font size will be 20px
  </div>
</div>
`;

exports[`cutest use glamor helpers 1`] = `
<div
  data-css-snd14h=""
>
  <div
    className="one"
  >
    <div
      className="two"
    >
      <div
        className="three"
      >
        <span
          className="inner"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;
